src/core/utils/Range.hpp

```diff
@@ -0,0 +1,43 @@
+#ifndef UTILS_RANGE_HPP
+#define UTILS_RANGE_HPP
+
+#include <utility>
+
+namespace Utils {
+
+/**
+ * @brief Helper class that wraps two iterators into
+          a range object.
+*/
+template <typename Iterator> class Range {
```

Comment:
I think the way to go if we want to use any range library would be ranges v3 from eric niebler, because I looks like some form of that one is going to be standarized (bosot ranges is ranges v2). Not non of this is clear yet, and ranges v3 needs C++14. So I think we can live with these 10 lines of trivial code for now.